api/models.py

```diff
@@ -349,24 +349,16 @@ class Regression(Node):
         description='Regression details'
     )
 
-    _OBJECT_ID_FIELDS = [
+    _OBJECT_ID_FIELDS = Node._OBJECT_ID_FIELDS + [
         'regression_data.parent',
```

Comment:
Actually, there's no field called `regression_data.parent` as `regression_data` is a list of nodes.  So I'm not sure how this is working.  Looking at the code:

```python
    @classmethod
    def _translate_object_ids(cls, params):
        for key, value in params.items():
            if key in cls._OBJECT_ID_FIELDS:
                yield key, ObjectId(value)
```
with `params` being a dictionary typically received from a query.  If a parameter is `parent=1234` then `1234` will be converted to an ObjectId and that's then used to look up documents in the database.  I think this is fine.  However, if a parameter is `regression_data.parent=1234` then we'll be looking for regression documents with `regression_data.parent=ObjectId("1234")` but there won't be any in the database as what we actually have is objects with `regression_data[0].parent=ObjectId("1234")`.  Also the regression data is a list of nodes so the list of fields to translate is the same as for a Node object, it's just being repeated here.

Does this actually work?  Is it possible to find regressions based on e.g. the parent id of a node listed in `regression_data`?

Reply:
Yes, it does work. 
Typically, filters for `regression.data.field_name` works just like `revision.field_name`.
Translation for Node fields inside `regression_data` won't work from `Node.translate_fields` as it will try to compare field names and it won't be the same for regression data as it will have `regression_data.` prefix.
That is why we need to override it in the `Regression` model.

Reply:
Sample output from my local setup where I have 6 total regression nodes in the DB:
```
$ curl 'http://localhost:8001/latest/nodes?kind=regression&regression_data.parent=6436a645ef2451d1c50dd3d6'
{"items":[{"id":"64476ff0b1dc0e9fff806237","kind":"regression","name":"kver","path":["checkout","kver"],"group":"kver","revision":{"tree":"kernelci","url":"https://github.com/kernelci/linux.git","branch":"staging-mainline","commit":"49952f28d24e337e064f53e5e8568cd88a5dcb76","describe":"staging-mainline-20230412.0","version":{"version":6,"patchlevel":3,"sublevel":null,"extra":"-rc6-35-g49952f28d24e","name":null}},"parent":"64476ff0b1dc0e9fff806236","state":"done","result":"fail","artifacts":{"tarball":"http://172.17.0.1:8002/linux-kernelci-staging-mainline-staging-mainline-20230412.0.tar.gz"},"data":null,"created":"2023-04-25T06:15:12.931000","updated":"2023-04-25T06:15:12.931000","timeout":"2023-04-25T12:15:12.931000","holdoff":null,"regression_data":[{"id":"6436a678ef2451d1c50dd3da","kind":"node","name":"kver","path":["checkout","kver"],"group":"kver","revision":{"tree":"kernelci","url":"https://github.com/kernelci/linux.git","branch":"staging-mainline","commit":"49952f28d24e337e064f53e5e8568cd88a5dcb76","describe":"staging-mainline-20230412.0","version":{"version":6,"patchlevel":3,"sublevel":null,"extra":"-rc6-35-g49952f28d24e","name":null}},"parent":"6436a645ef2451d1c50dd3d6","state":"done","result":"pass","artifacts":{"tarball":"http://172.17.0.1:8002/linux-kernelci-staging-mainline-staging-mainline-20230412.0.tar.gz"},"data":null,"created":"2023-04-12T12:39:20.946000","updated":"2023-04-12T12:39:31.609000","timeout":"2023-04-12T18:39:20.946000","holdoff":null},{"id":"64476ff0b1dc0e9fff806236","kind":"node","name":"kver","path":["checkout","kver"],"group":"kver","revision":{"tree":"kernelci","url":"https://github.com/kernelci/linux.git","branch":"staging-mainline","commit":"49952f28d24e337e064f53e5e8568cd88a5dcb76","describe":"staging-mainline-20230412.0","version":{"version":6,"patchlevel":3,"sublevel":null,"extra":"-rc6-35-g49952f28d24e","name":null}},"parent":"6436a645ef2451d1c50dd3d6","state":"done","result":"fail","artifacts":{"tarball":"http://172.17.0.1:8002/linux-kernelci-staging-mainline-staging-mainline-20230412.0.tar.gz"},"data":null,"created":"2023-04-25T06:15:12.909000","updated":"2023-04-25T06:15:12.909000","timeout":"2023-04-25T12:15:12.909000","holdoff":null}]}],"total":1,"limit":50,"offset":0}
```

Reply:
OK I've imported the returned object into my local database and then was able to find it again with this query:
```
> db.node.findOne({'regression_data.group': 'kver'})
{
	"_id" : ObjectId("645263dd084eebe916b1be7d"),
	"id" : "64476ff0b1dc0e9fff806237",
	"kind" : "regression",
	"name" : "kver",
	"path" : [
		"checkout",
		"kver"
	],
	"group" : "kver",
	"revision" : {
		"tree" : "kernelci",
		"url" : "https://github.com/kernelci/linux.git",
		"branch" : "staging-mainline",
		"commit" : "49952f28d24e337e064f53e5e8568cd88a5dcb76",
		"describe" : "staging-mainline-20230412.0",
		"version" : {
			"version" : 6,
			"patchlevel" : 3,
			"sublevel" : null,
			"extra" : "-rc6-35-g49952f28d24e",
			"name" : null
		}
	},
	"parent" : "64476ff0b1dc0e9fff806236",
	"state" : "done",
	"result" : "fail",
	"artifacts" : {
		"tarball" : "http://172.17.0.1:8002/linux-kernelci-staging-mainline-staging-mainline-20230412.0.tar.gz"
	},
	"data" : null,
	"created" : "2023-04-25T06:15:12.931000",
	"updated" : "2023-04-25T06:15:12.931000",
	"timeout" : "2023-04-25T12:15:12.931000",
	"holdoff" : null,
	"regression_data" : [
		{
			"id" : "6436a678ef2451d1c50dd3da",
			"kind" : "node",
			"name" : "kver",
# ...
```
(small glitch with `id` vs `_id` but that's a detail).

I haven't searched enough in the MongoDB documentation to find the part about this behaviour, but here's a community discussion about it:
https://www.mongodb.com/community/forums/t/how-to-query-an-array-of-objects-to-match-any-of-the-provided-values/159185/9

I think this is slightly unexpected and other database engines might not be able to do that, so small things like this kind of make us tied to MongoDB.  It would be good to know if other NoSQL databases have similar features.  In any case, I guess we're OK with this for now.

As a small improvement, I still think it would be good to not have to repeat all the fields from Node that need to be translated in `regression_data`.  But that could be a follow-up thing.

Reply:
It seems like you inserted `id` field as well while creating `regression` node. And then it shows two IDs. This is weird.

Reply:
No it's normal, I just dumped the output from your sample `curl` command which had an `id` attribute into my MongoDB instance and then of course it created a new `_id` field for the document.